contracts/cw-ics721-bridge/src/msg.rs

```diff
@@ -0,0 +1,189 @@
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::IbcTimeout;
+use cw721_proxy_derive::cw721_proxy;
+use cwd_interface::ModuleInstantiateInfo;
+
+#[cw_serde]
+pub struct InstantiateMsg {
+    /// Code ID of cw721-ics contract. A new cw721-ics will be
+    /// instantiated for each new IBCd NFT classID.
+    ///
+    /// NOTE: this _must_ correspond to the cw721-base contract. Using
+    /// a regular cw721 may cause the ICS 721 interface implemented by
+    /// this contract to stop working, and IBCd away NFTs to be
+    /// unreturnable as cw721 does not have a mint method in the spec.
+    pub cw721_base_code_id: u64,
+    /// An optional proxy contract. If a proxy is set the contract
+    /// will only accept NFTs from that proxy. The proxy is expected
+    /// to implement the cw721 proxy interface defined in the
+    /// cw721-proxy crate.
+    pub proxy: Option<ModuleInstantiateInfo>,
```

Comment:
It's really nice to encapsulate contract info like this. However, `cwd_interface` isn't generic enough and has references to proposals and `dao_address`. This has nothing to do with ics721. Also the term "module" is a bit confusing since it is nothing more than a regular contract. 

I'd rather pull this into a package in this repo called `cw-ics721`. Maybe the struct can be called `ContractInstantiateInfo` instead. It's even something that would be useful to get into cw-plus or cw-utils.

Reply:
good point. added a `cw-cii` (cosmwasm contract instantiate info) package and moved the logic into there.